Standardize development environment on validated working changes

justwheel opened the pull request:
This PR includes two commits, explained below:

---

**Dockerfile: Install mysqlclient library in previous layer** (b03ed0c098bfe08ffb6a75be2cf949a5ecc34851)

This moves the `pip3 install` step for the Docker container into the
same layer as system dependencies. This way, the cached version of
mysqlclient is used whenever our project dependencies change.

---

**docker-compose/docs: Run app as single container** (eb9a9e8d7aa7b2d83e78e017ebe638122ca8baa2)

This commit makes changes based on in-person testing and feedback
collected from @ldidonato and @leong96 to get a working Docker Compose
environment working on multiple operating systems.

The docker-compose.yml is reverted to an earlier version. The previous
version created five versions of the application in parallel. This is
hard to test since each team member is using a specific environment to
them. Instead, this commit creates a common environment for everyone to
depend on for the app to work. Doing it this way allows us, as a group,
to quickly be aware if there is a breaking change made to the app
(because it will break for everyone instead of just one person). We
discovered some of this difficulty when debugging with Lauren, Harrison,
and I today.

Lastly, I also updated the docs for the changed port address of running
the application. I changed the port because it is an uncommon port and
it appeared macOS was already doing something on port 8000 and blocked
the app from starting. Using port 16000 for local development was a
random port pick to avoid battling with conflicting ports.

justwheel committed:
Dockerfile: Install mysqlclient library in previous layer

This moves the `pip3 install` step for the Docker container into the
same layer as system dependencies. This way, the cached version of
mysqlclient is used whenever our project dependencies change.

Signed-off-by: Justin W. Flory <git@jwf.io>

justwheel committed:
docker-compose/docs: Run app as single container

This commit makes changes based on in-person testing and feedback
collected from @ldidonato and @leong96 to get a working Docker Compose
environment working on multiple operating systems.

The docker-compose.yml is reverted to an earlier version. The previous
version created five versions of the application in parallel. This is
hard to test since each team member is using a specific environment to
them. Instead, this commit creates a common environment for everyone to
depend on for the app to work. Doing it this way allows us, as a group,
to quickly be aware if there is a breaking change made to the app
(because it will break for everyone instead of just one person). We
discovered some of this difficulty when debugging with Lauren, Harrison,
and I today.

Lastly, I also updated the docs for the changed port address of running
the application. I changed the port because it is an uncommon port and
it appeared macOS was already doing something on port 8000 and blocked
the app from starting. Using port 16000 for local development was a
random port pick to avoid battling with conflicting ports.

Signed-off-by: Justin W. Flory <git@jwf.io>

nlMeminger commented on docker-compose.yml:
```diff
@@ -1,68 +1,19 @@
-      - db
-    ports:
-      - 9000:8000
+      - 16000:8000
```
The port on the host should be 8000, not 16000.  There are already firewall rules for 8000.  Switching to 16000 would require more work

nlMeminger commented on docs/dev-environment.md:
```diff
@@ -53,4 +53,4 @@ To shut down `docker-compose` in detached mode, use this command:
 
 Once `docker-compose` is running, open a web browser.
-Visit [localhost:8000](http://localhost:8000/) to view the site running locally.
+Visit [localhost:16000](http://localhost:16000/) to view the site running locally.
```
again, should be port 8000

justwheel committed:
docker-compose: Run app on port 8000 on localhost

This commit undoes a change in a previous commit to run the app on port
16000. Admittedly it is a non-standard port and in the case of this
morning, another container was likely already running on the port.

Docs are also updated.

cc: @nlMeminger

Signed-off-by: Justin W. Flory <git@jwf.io>
